File: cpp/01-commands/tests/SumOfDigitsKata/Number.cpp
#include "pch.h"
#include "Number.h"

Number::Number(long number)
    : _number(number)
{}

long Number::DigitalRoot()
{
    long result = 0;

    while (_number > 0)
    {
        result += _number % 10;
        _number = _number / 10;
    }

    if (result > 9)
    {
        Number number(result);
        result = number.DigitalRoot();
    }

    return result;
}

File: python/01-commands/tests/SumOfDigitsKata/number.py
class Number:
    def __init__(self, number):
        self._number = number

    def digital_root(self):
        result = 0

        while self._number > 0:
            result += self._number % 10
            self._number = self._number // 10

        if result > 9:
            number = Number(result)
            result = number.digital_root()

        return result